tests/xmap_test.py

```diff
@@ -272,11 +276,204 @@ def tearDown(self):
     jax.experimental.maps.EXPERIMENTAL_SPMD_LOWERING = self.old_lowering_flag
 
 
+LHSSpec = RHSSpec = Tuple[int, ...]
+DimSpec = Tuple[LHSSpec, RHSSpec]
+
+class PdotSpec(NamedTuple):
+  """Specifies positional axes to contract or batch via mapping or position."""
+  map_contract: DimSpec
+  pos_contract: DimSpec
```

Comment:
Yes, it's about mapped axes and uses positions. All the info in PdotSpec uses positional axes, in this case to identify which positional axes xmap should map. That is, the fields `map_contract` and `map_batch` are accessed (indirectly, via `lhs_mapped` and rhs_mapped` when setting up the `xmap` call itself, rather than the `pdot` call.

The properties `pos_contract_after_mapping` and `pos_batch_after_mapping` compute the positions _after_ mapping has eliminated some.

Does that make sense?

Reply:
I revised the data structure which might things clearer, though I still stuck with specifying things in terms of positional axes.

Reply:
Yeah, I think that what's there now is great. Thanks!